Source/Scene/PostProcessorScene.js

```diff
@@ -20,23 +20,247 @@ define([
         this.blackAndWhiteStage = createBlackAndWhiteStage();
+        this.blackAndWhiteStage.show = false;
         this.brightnessStage = createBrightnessStage();
+        this.brightnessStage.show = false;
         this.eightBitStage = createEightBitStage();
+        this.eightBitStage.show = false;
         this.compositeTextureStage = createCompositeTextureStage();
         this.compositeTextureStage.show = false;
+        this.LensFlareStage = createLensFlareStage();
 
         var stages = [
             this.blackAndWhiteStage,
             this.brightnessStage,
             this.eightBitStage,
-            this.compositeTextureStage
+            this.compositeTextureStage,
+            this.LensFlareStage
         ];
 
         this._postProcessor = new PostProcessor({
             stages : stages
         });
     }
 
+    function createLensFlareStage() {
+        var url_Dirt = buildModuleUrl('Assets/Textures/LensFlare/DirtMask.jpg');
+        var url_Star = buildModuleUrl('Assets/Textures/LensFlare/StarBurst.jpg');
+
+        var uniformValues = {
+
+            DirtTexture: url_Dirt,
+            StarTexture: url_Star,
+
+            Intensity: 2.0,
+
+            Distortion: 5.0,
+            NumofGhosts: 4,
+            GhostDispersal: 0.4,
+
+            HaloWidth: 0.4,
+
+            EarthRadius: 0.0
+        };
+
+        var LensFlareShader =
+            'uniform sampler2D u_colorTexture; \n' +
+            'uniform sampler2D u_DirtTexture; \n' +
+            'uniform sampler2D u_StarTexture; \n' +
+
+            'uniform float u_Distortion; \n' +
+            'uniform float u_GhostDispersal; \n' +
+            'uniform float u_HaloWidth; \n' +
+            'uniform float u_EarthRadius; \n' +
+
+            'uniform float u_Intensity; \n' +
+
+            'varying vec2 v_textureCoordinates; \n' +
+
+            //return ndc from world coordinate biased EarthRadius
+            'vec4 GetNDCFromWC(in vec3 WC, in float EarthRadius) \n' +
+            '{ \n' +
+            ' vec4 position = vec4(WC, 1.0); \n' +
+            ' vec4 positionEC = czm_view * position; \n' +
+            ' positionEC = vec4(positionEC.x + EarthRadius , positionEC.y, positionEC.z, 1.0); \n' +
+            ' vec4 positionWC = czm_eyeToWindowCoordinates(positionEC); \n' +
+            ' return czm_viewportOrthographic * vec4(positionWC.xy, -positionWC.z, 1.0); \n' +
+            '} \n' +
+
+            //Check if current pixel is included Earth
+            //if then mask it gradually
+            'float IsinEarth(in vec2 texcoord, in vec2 SceneSize) \n' +
+            '{ \n' +
+
+                'vec2 NDC = texcoord * 2.0 - 1.0; \n' +
+                'vec4 EarthPosSC = GetNDCFromWC(vec3(0.0,0.0,0.0), 0.0); \n' +
+                'vec4 EarthPosSCEdge = GetNDCFromWC(vec3( 0.0,0.0,0.0), u_EarthRadius *1.5); \n' +
+
+                'NDC.xy -= EarthPosSC.xy; \n' +
+                'float X =  abs(NDC.x)*SceneSize.x;   \n' +
```

Comment:
The GLSL could use a lot of tweaks like this to look like our other GLSL code.  Feel free to make them now or wait until this is in a separate .glsl file where it would be easier and the WebStorm auto formatter might work (we use it for JavaScript).